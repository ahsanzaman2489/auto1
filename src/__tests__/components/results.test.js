import React from 'react';
import ResultsComponent from '../../components/results';
import {shallow} from 'enzyme';
import renderer from "react-test-renderer";


const setup = (props = {}, state = null) => {
    const defaultProps = {totalItemCount: 20, currentPage: 2, currentItemCount: 10, itemPerPage: 10};
    const setUpProps = {...defaultProps, ...props};
    const wrapper = shallow(<ResultsComponent {...setUpProps}/>);
    const tree = () => renderer.create(<ResultsComponent {...setUpProps}/>).toJSON();
    if (state) wrapper.setState(state);
    return {wrapper, props: setUpProps, tree}
};


describe("Results component", () => {
    it('should match with snapshot', () => {
        const {tree} = setup();
        expect(tree()).toMatchSnapshot();
    });

    it('should render results text with values', () => {
        const {wrapper, props} = setup();
        expect(wrapper.find('.searchResult').text()).toBe(`Showing ${(props.itemPerPage * (props.currentPage - 1)) + props.currentItemCount} of ${props.totalItemCount} results`);
    });
});


