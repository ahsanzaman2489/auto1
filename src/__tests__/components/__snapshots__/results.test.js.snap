// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results component should match with snapshot 1`] = `
<div
  className="searchResult"
>
  <p>
    Showing 
    20
     of 
    20
     results
  </p>
</div>
`;
